src/Components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,32 @@
+import { useContext } from 'react';
+import { TodoContext } from '../TodoContext/TodoContext';
+import { TodoListItem } from '../Todo/TodoListItem';
+import { Todo } from '../../types/Todo';
+
+type FilterSettings = string;
+
+function filterList(list: Todo[], settings: FilterSettings): Todo[] {
+  return [...list].filter(item => {
```

Comment:
```suggestion
  return list.filter(item => {
```
list doesn't mutate the array